lib/rules/no-deprecated-v-bind-sync.js

```diff
@@ -0,0 +1,51 @@
+/**
+ * @author Przemyslaw Falowski (@przemkow)
+ * @fileoverview Disallow use of deprecated `.sync` modifier on `v-bind` directive (in Vue.js 3.0.0+)
+ */
+'use strict'
+
+// ------------------------------------------------------------------------------
+// Requirements
+// ------------------------------------------------------------------------------
+
+const utils = require('../utils')
+
+// ------------------------------------------------------------------------------
+// Rule Definition
+// ------------------------------------------------------------------------------
+
+module.exports = {
+  meta: {
+    type: 'problem',
+    docs: {
+      description: 'disallow use of deprecated `.sync` modifier on `v-bind` directive (in Vue.js 3.0.0+)',
+      category: undefined,
+      url: 'https://eslint.vuejs.org/rules/no-deprecated-v-bind-sync.html'
+    },
+    fixable: 'code',
+    schema: [],
+    messages: {
+      syncModifierIsDeprecated: "'.sync' modifier on 'v-bind' directive is deprecated. Use 'v-model:propName' instead."
+    }
+  },
+  create (context) {
+    return utils.defineTemplateBodyVisitor(context, {
+      "VAttribute[directive=true][key.name.name='bind']" (node) {
+        if (node.key.modifiers.map(mod => mod.name).includes('sync')) {
+          context.report({
+            node,
+            loc: node.loc,
+            messageId: 'syncModifierIsDeprecated',
+            fix: (fixer) => {
+              const isUsingSpreadSyntax = node.key.argument == null
+              if (isUsingSpreadSyntax) {
+                return
+              }
+              return fixer.replaceText(node.key, `v-model:${node.key.argument.rawName}`)
```

Comment:
Maybe this doesn't work for dynamic argument.
Building the fixed source code using `context.getSourceCode(). getText(node.key.argument)` may work.

https://github.com/mysticatea/vue-eslint-parser/blob/master/docs/ast.md#vdirectivekey

Reply:
You are completely right, I totally missed the fact that `VDirectiveKey` supports not only `VIdentifier` but also `VExpressionContainer`. PR updated, dynamic argument should work correctly now.